miden-lib/src/transaction/mod.rs

```diff
@@ -180,7 +180,7 @@ impl TransactionKernel {
         let storage_root = account_header.storage_commitment();
         let code_root = account_header.code_commitment();
         let account_key =
-            Digest::from([account_id.first_felt(), account_id.second_felt(), ZERO, ZERO]);
+            Digest::from([account_id.second_felt(), account_id.first_felt(), ZERO, ZERO]);
```

Comment:
What is the motivation for this reversal?

Also, we should probably document where this order is important (I think `start_foreign_context` procedure relies on this, and maybe some procedure that gets executed as a part of transaction prologue?)

Reply:
This change addresses this comment: https://github.com/0xPolygonMiden/miden-base/pull/982#discussion_r1897323636

So yes, exactly, this is (for now) only important for `start_foreign_context`. The prologue doesn't currently "rely" on this function as the native account's data is loaded slightly differently.

Added a note to document the link between this function and the procedure.